inference-engine/src/transformations/include/transformations/utils/utils.hpp

```diff
@@ -48,8 +48,15 @@ bool has_op_with_type(const std::shared_ptr<const ngraph::Function> &function) {
 inline std::string create_ie_output_name(const ngraph::Output<ngraph::Node>& output) {
     const auto& prev_layer = output.get_node_shared_ptr();
     std::string out_name = prev_layer->get_friendly_name();
-    if (prev_layer->get_output_size() != 1)
+    size_t num_outputs(0);
+    for (const auto& out : prev_layer->outputs()) {
+        if (!out.get_target_inputs().empty())
+            num_outputs++;
+    }
+    if (num_outputs > 1 || output.get_target_inputs().empty())
```

Comment:
if `num_outputs == 1`, but `output.get_target_inputs().empty() == true`, `out_name `will be `friendly_name + idx + ".fake"`,  is it expected? why we should add idx?

Reply:
In order to save unique names we should add index. `fake` postfix I added only to mark this outputs without consumers.

Reply:
"In order to save unique names we should add index." - but `num_outputs == 1`, it will be already unique without idx.
Now we have such logic (for not fake outputs): if we have only one output, we do not add idx. But for fakes it will be different - we add idx anyway.

Reply:
cloneNet creates all output data pointers. So we don't add such outputs but create data.